pkg/backup/item_backupper.go

```diff
@@ -253,11 +253,31 @@ func (ib *defaultItemBackupper) backupItem(logger logrus.FieldLogger, obj runtim
 		return false, kubeerrs.NewAggregate(backupErrs)
 	}
 
+	// group version of this object
+	// Used on filepath to backup up all groups and versions
+	version := resourceVersion(obj)
+
+	// Getting the preferred group version of this resource
```

Comment:
we might want to consider moving the getting of the preferred version to an outer loop (e.g. in `resource_backupper.go`) - there's no need to do this once for every item, we could just do it once per resource type. I'm not sure if there's a material performance impact or not.